src/feed_parser.c

```diff
@@ -128,18 +130,20 @@ feed_parser_auto_discover (feedParserCtxtPtr ctxt)
 	/* FIXME: we only need the !g_str_equal as a workaround after a 404 */
 	if (source && !g_str_equal (source, subscription_get_source (ctxt->subscription))) {
 		debug1 (DEBUG_UPDATE, "Discovered link: %s", source);
-		ctxt->failed = FALSE;
 		subscription_set_source (ctxt->subscription, source);
 
 		/* The feed that was processed wasn't the correct one, we need to redownload it.
 		 * Cancel the update in case there's one in progress */
 		subscription_cancel_update (ctxt->subscription);
 		subscription_update (ctxt->subscription, FEED_REQ_RESET_TITLE);
 		g_free (source);
-	} else {
-		debug0 (DEBUG_UPDATE, "No feed link found!");
-		g_string_append (ctxt->feed->parseErrors, _("The URL you want Liferea to subscribe to points to a webpage and the auto discovery found no feeds on this page. Maybe this webpage just does not support feed auto discovery."));
+
+		return TRUE;
 	}
+
+	debug0 (DEBUG_UPDATE, "No feed link found!");
+	g_string_append (ctxt->feed->parseErrors, _("The URL you want Liferea to subscribe to points to a webpage and the auto discovery found no feeds on this page. Maybe this webpage just does not support feed auto discovery."));
+	return FALSE;
 }
```

Comment:
I had not noticed the implicit return here before, in the sea of mostly useless warnings. Hopefully the compiler would have complained louder if we had tried to use the value.

Reply:
True the warnings are a hassle. I'm trying to reduce them more and more.